pkgs/dart_model/lib/src/json_buffer/iterables.dart

```diff
@@ -39,3 +39,13 @@ class _IteratorFunctionIterable<T> extends Iterable<T> {
   @override
   Iterator<T> get iterator => _function();
 }
+
+/// A `Map` in a `JsonBufferBuilder`.
+abstract interface class MapInBuffer {
```

Comment:
Fwiw we could also use Expandos for this, which would also allow us to record parents for other types of maps (and wouldn't need this extra interface and associated type checks).

Reply:
Good point, thanks.

Another way to tweak would be to lean on the type more, the parent of a map in a buffer is definitely also a map in a buffer. So there should only be one type check to get the whole path. (And as noted elsewhere, we should store the key used as well as the parent, so it really is fast).